android/src/com/mapswithme/maps/Framework.java

```diff
@@ -61,6 +61,13 @@
     public boolean buildings;
   }
 
+  public static class RouteAltitudeLimits
+  {
+    public int minRouteAltitude;
```

Comment:
What's the problem to get current Route, and create a couple of methods, like:
```c++
GenerateChart(Route const & route, ...)
GetMinMaxAltitude(Route const & route, ...)
```
and then call both methods on the same route instance?